### ISSUE TITLE
Plugin fails with upcoming Maven 3.9.0

### ISSUE THREAD

**Issue by cstamas** [revelc/impsort-maven-plugin#64]
**Describe the bug**
The impsort-maven-plugin depends on a Maven2 feature that is about to be removed in Maven 3.9,0 and Maven 4.0,0: "auto injection" of plexus-utils. In short: plugin uses but does NOT declares COMPILE dependency on plexus-utils. Maven2 was making plexus-utils "provided", while Maven 3.x supported this for backward compatilibty reasons, but Maven 3,9 and later will not.

**Versions (OS, Maven, Java, and others, as appropriate):**
 - Affected version(s) of this project: all
 - OS: all
 - Others: 

**To Reproduce**
1. build locally Maven 3.9 or 4.0 snapshot
2. build some project using this plugin (for example maven daemon)
3. See error
```
[ERROR] Failed to execute goal net.revelc.code:impsort-maven-plugin:1.7.0:sort (sort-imports) on project mvnd-build-maven-plugin: Execution sort-imports of goal net.revelc.code:impsort-maven-plugin:1.7.0:sort failed: A required class was missing while executing net.revelc.code:impsort-maven-plugin:1.7.0:sort: null
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>net.revelc.code:impsort-maven-plugin:1.7.0
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/home/cstamas/.m2/repository-oss/net/revelc/code/impsort-maven-plugin/1.7.0/impsort-maven-plugin-1.7.0.jar
[ERROR] urls[1] = file:/home/cstamas/.m2/repository-oss/com/github/javaparser/javaparser-core/3.24.2/javaparser-core-3.24.2.jar
[ERROR] urls[2] = file:/home/cstamas/.m2/repository-oss/com/google/guava/guava/31.1-jre/guava-31.1-jre.jar
[ERROR] urls[3] = file:/home/cstamas/.m2/repository-oss/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar
[ERROR] urls[4] = file:/home/cstamas/.m2/repository-oss/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
[ERROR] urls[5] = file:/home/cstamas/.m2/repository-oss/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar
[ERROR] urls[6] = file:/home/cstamas/.m2/repository-oss/org/checkerframework/checker-qual/3.12.0/checker-qual-3.12.0.jar
[ERROR] urls[7] = file:/home/cstamas/.m2/repository-oss/com/google/errorprone/error_prone_annotations/2.11.0/error_prone_annotations-2.11.0.jar
[ERROR] urls[8] = file:/home/cstamas/.m2/repository-oss/com/google/j2objc/j2objc-annotations/1.3/j2objc-annotations-1.3.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------
```

**Expected behavior**
To work as before.

**Comment by ctubbsii** [comment]
For what it's worth, every PR in GitHub is also an issue. If you're going to immediately submit a PR, it's not strictly necessary to create an issue first. It's not a huge problem, and there are good reasons to do it sometimes, but it does double up the issue tracker, so just wanted to mention you don't have to do it for this project.

**Comment by gnodet** [comment]
@ctubbsii Do you plan a release soon ? Maven 4.0.0-alpha-2 has been released already and exhibits the same problem, which makes it unusable in any project using this plugin.

**Comment by ctubbsii** [comment]
@gnodet I released today.

**Comment by hazendaz** [comment]
I'd suggest using jdk 11 to build.  Your product will still target java 8 or whatever you told it.  There is little to no reason to use toolchains or continue using java 8.  There are already a good number of plugins that require java 11 and stopped supporting java 8.  Additionally, if it's a must to stay where you are, then it would be best to stay on maven 3.8.x which is still supported.

I cannot speak for final decision in the end but don't see the value of trying to continue jdk 8 support when it's so well proven unnecessary from a build perspective.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by hazendaz** [comment]
And of course you have a simple workaround.  Many plugins need the same especially if not typical usage or no longer supported.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ctubbsii** [comment]
Because there are several workarounds: using newer Java 11 to build, but with `--release 8` if you need Java 8 as a target, or your workaround explicitly adding plexus-utils as a direct dependency, I don't see much reason to patch and release an older version.

**Comment by hazendaz** [comment]
@sbernard31 Its not longer an issue cross compiling these days to run into edge cases.  However, the release option doesn't do anything with transient dependencies so you would additionally want enforcer plugin to make sure its binary compatible.  Other than that its pretty solid.  Speaking for myself and my daily life, we run something like 2k builds with more than 80% on jdk 17 but reality is varied older jdk usage from 8 to 11 and no issues doing so.

On maven suggestion, only indicating maven 3.9.0 isn't entirely necessary.  That is for compatibility checks in moving to maven 4.  It works well, work around needed in this particular case but maven is still releasing 3.8.x line.  In fact, believe 3.8.8 along with 3.9.1 are going to voting likely later this week.  So 3.8.x isn't dead yet.  That was my only point in regards there.  In my daily life we moved to 3.9.0 but we did hit some edge cases and even there worked around any we found.
